ms2query/networking.py

```diff
@@ -58,24 +62,247 @@ def add_library_connections(graph, similarity_matrix, lib_ids):
+            color=np.array(node_type).astype(int),
+            colorscale='portland',
+            line_color="white",
+            line_width=2))
+
+    layout = go.Layout(
+        autosize=True,
+        width=700,
+        height=600,
+        paper_bgcolor='rgba(0,0,0,0)',
+        plot_bgcolor='rgba(0,0,0,0)',
+
+        xaxis=go.layout.XAxis({'showgrid': False,
+                               'visible': False, }),
+        yaxis=go.layout.YAxis({'showgrid': False,
+                               'visible': False, }),
+        margin=go.layout.Margin(
+            l=50,
+            r=50,
+            b=50,
+            t=50,
+            pad=4,
+        )
+    )
+
+    fig = go.Figure(data=edge_trace + [node_trace],
+                    layout=layout)
+    return fig
+
+
+def make_plotly_edge(u: Union[str, int],
+                     v: Union[str, int],
+                     d: dict,
+                     pos: dict,
+                     attribute: str,
+                     width_default: float,
+                     style: str = "solid",
+                     max_val: Union[float, int] = 1,
+                     cmap: Union[None, colors.Colormap] = None):
+    """Returns go.Scatter for the edge object
+
+    Args:
+    -------
+    u:
+        Starting node of edge.
+    v:
+        End node of edge.
+    d:
+        Attribute data of node
+    pos: dict of {node: [x,y]}
+        Contains coordinates for each node
+    attribute: str
+        The attribute that should be looked at for the score
+    width_default:
+        The max width of an edge
+    style:
+        The style of the edges. Default = solid
+    max_val:
+        Maximum value of edge attribute.
+    cmap:
+        If provided, a cmap to colour the edges by attribute score.
+        Default = none, meaning #888 will be used as a colour
+    """
+    x0, y0 = pos[u]
+    x1, y1 = pos[v]
+    xs = (x0, x1, None)
+    ys = (y0, y1, None)
+    # text coordinates
+    x_txt = [(x0 + x1) / 2]
+    y_txt = [(y0 + y1) / 2]
+    # default is the library connection parameters
+    val = d[attribute]
+    cor_val = val / max_val
+    if isinstance(val, bool):
+        val = int(val)
+    e_width = width_default * cor_val
+    if e_width == 0.0:
+        e_width = 0.8  # for mass_match==0
+    if cmap:
+        e_colour = colors.to_hex(cmap(cor_val))
+    else:
+        e_colour = "#888"
+    edge_trace = go.Scatter(
+        x=xs,
+        y=ys,
+        line=dict(width=e_width, color=e_colour, dash=style),
+        mode='lines')
+    # edge_text = ["{}: {}".format(attribute, str(d[attribute]))]
+    # edge_trace.text = edge_text
+
+    txt_trace = go.Scatter(
+        x=x_txt,
+        y=y_txt,
+        customdata=[["{}: {:.3f}".format(attribute, val)]],
+        mode='text',
+        hovertemplate="%{customdata[0]}<extra></extra>",
+        showlegend=False)
+
+    return edge_trace, txt_trace
 
 
 def draw_edges(network, pos, attribute, attr_cutoff, width_default,
```

Comment:
That function is not longer needed, right?

Reply:
Indeed. I'll remove it after merging the legend plotting branch :)